server/views/account/signin.jade

```diff
@@ -23,3 +23,29 @@ block content
         p
             a(href="/email-signin") If you originally signed up using your email address, you can sign in here.
 
+    script.
+      $(document).ready(function() {
+        var method = localStorage.getItem('lastSigninMethod'),
+          btnSelector = 'a.btn.btn-lg.btn-block.btn-social';
+        if (method) {
+          var x = JSON.parse(method), Class = x.methodClass, link = x.methodLink;
+          $.each($(btnSelector), function(i, item) {
```

Comment:
You'll need to wrap this parse method in a try-catch. I don't think we need to handle the application error but we should least be aware that an error can happen here.

Also, if we are going to use multiple assignments, lets keep them on separate lines.